lib/App/index.js

```diff
@@ -56,16 +58,72 @@ const FLOW_TYPES = [ 'triggers', 'conditions', 'actions' ];
 class App {
 
 	constructor( appPath ) {
-		this.path = path.resolve(appPath);
-		this._app = new HomeyLibApp( this.path );
+    this.path = path.resolve(appPath);
+    this.srcPath = this.path;
+    this._app = new HomeyLibApp( this.path );
     this._appJsonPath = path.join( this.path, 'app.json' );
     this._pluginsPath = path.join( this.path, '.homeyplugins.json');
     this._homeyComposePath = path.join(this.path, '.homeycompose');
-		this._exiting = false;
-		this._std = {};
+    this._exiting = false;
+    this._std = {};
     this._git = new GitCommands(appPath);
   }
 
+
+  /* 
+  * Returns true if typescript is set to true in the app.json 
+  */
+  async usesTypeScript() {
+    const json = await this.getAppJson();
+    return json.typescript;
+  }
+
+  /* 
+  * Returns the json object found in app.json 
+  */
+  async getAppJson() {
+    let hasCompose = await this._hasPlugin('compose');
+    let appJsonPath;
+
+    if(hasCompose) {
+      let appJsonComposePath = path.join(this.path, '.homeycompose', 'app.json');
+      let exists = false;
+      try {
+        await accessAsync(appJsonComposePath, fs.constants.R_OK | fs.constants.W_OK);
+        exists = true;
+      } catch( err ) {}
+
+      if( exists ) {
+        appJsonPath = appJsonComposePath;
+      } else {
+        appJsonPath = path.join(this.path, 'app.json');
+      }
+    } else {
+      appJsonPath = path.join(this.path, 'app.json');
+    }
+        
+    const appJson = await readFileAsync(appJsonPath, 'utf8');
+    return JSON.parse(appJson);
+  }
+
+  async convertFromTypeScript(appPath) {
+    Log(colors.green('✓ Typescript detected. Running tsc...'));
+  
+    try {
+      const newPath = path.join(appPath, 'ts-output/');
+      await fse.remove(newPath).catch(err => {});
+      await exec(`cd ${appPath} && tsc`);
```

Comment:
This will only work if `tsc` is installed globally right? I think we should try to run `node_modules/.bin/tsc`

Reply:
Yes, or `npx tsc` perhaps